From 1b51523c2f03edee073e6f9c97080a679b043388 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 22 Sep 2020 15:25:27 +0300
Subject: [PATCH] [SW-0] goya2: uart: pinmux config for legacy boot

UART pad select/pinmux configuration added for preboot
and ppboot goya2 asic platform for legacy boot flow
(without BTL).

Following are the pad/pinmux settings for uart,

GPIO_08 (UART 0 RXD, ALT FUNC1)
GPIO_09 (UART 0 TXD, ALT FUNC1)

These settings are already done in BTL code. So
these are not required in flow where BTL is run.

Legacy flow when BTL is not used i.e. booting ppboot/preboot
via JTAG, these settings are required.

Change-Id: I6e85d789efe9dfbce20c1cb665e0c5f6e80f2a3e
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/goya2/asic.c b/goya2/asic.c
index cf0f54a..ddf76f2 100644
--- a/goya2/asic.c
+++ b/goya2/asic.c
@@ -35,6 +35,8 @@
 	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
 					    HL_PLL_DIV3);
 
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
 	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
 }
 
